src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/credentials/CredentialMutation.kt

```diff
@@ -0,0 +1,96 @@
+package no.skatteetaten.aurora.gobo.graphql.credentials
+
+import com.expediagroup.graphql.spring.operations.Mutation
+import com.fasterxml.jackson.annotation.JsonProperty
+import graphql.schema.DataFetchingEnvironment
+import mu.KotlinLogging
+import no.skatteetaten.aurora.gobo.graphql.AccessDeniedException
+import no.skatteetaten.aurora.gobo.graphql.user.User
+import no.skatteetaten.aurora.gobo.integration.herkimer.CredentialBase
+import no.skatteetaten.aurora.gobo.integration.herkimer.HerkimerResult
+import no.skatteetaten.aurora.gobo.integration.herkimer.HerkimerService
+import no.skatteetaten.aurora.gobo.integration.herkimer.RegisterResourceAndClaimCommand
+import no.skatteetaten.aurora.gobo.integration.herkimer.ResourceKind
+import no.skatteetaten.aurora.gobo.security.checkValidUserToken
+import no.skatteetaten.aurora.gobo.security.currentUser
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+private val logger = KotlinLogging.logger {}
+
+@Component
+class CredentialMutation(
+    private val herkimerService: HerkimerService,
+    @Value("\${integrations.dbh.application.deployment.id}") val dbhAdId: String
+) : Mutation {
+    suspend fun registerPostgresMotelServer(
+        input: PostgresMotelInput,
+        dfe: DataFetchingEnvironment
+    ): RegisterPostgresResult {
+        dfe.checkValidUserToken()
+
+        dfe.currentUser().checkIsAuthorized()
+
+        val postgresInstance = input.toHerkimerPostgresInstance()
+
+        return herkimerService.registerResourceAndClaim(
+            RegisterResourceAndClaimCommand(
+                ownerId = dbhAdId,
+                credentials = postgresInstance,
+                resourceName = postgresInstance.instanceName,
+                claimName = "ADMIN",
+                resourceKind = ResourceKind.PostgresDatabaseInstance
+            )
+        ).toRegisterPostgresResult(input.host)
+    }
+}
+private fun User.checkIsAuthorized() {
+    if (!this.id.matches(Regex("system:serviceaccount:aurora[-\\w]*:vra$"))) {
+        throw AccessDeniedException("You do not have access to register a Postgres Motel Server")
+    }
+}
+
+private fun HerkimerResult.toRegisterPostgresResult(host: String): RegisterPostgresResult {
+    val message =
+        if (!this.success) "PostgresMotel host=$host could not be registered. The AuroraPlattform has internal configuration issues."
+        else "PostgresMotel host=$host has been successfully added."
+
+    return RegisterPostgresResult(
+        message,
+        this.success
+    )
+}
+
+private fun PostgresMotelInput.toHerkimerPostgresInstance() =
+    PostgresHerkimerDatabaseInstance(
+        host = host,
+        port = 5432,
+        // TODO: Should instanceName be a part of the input or should we deduce it?
+        instanceName = "${host.substringBefore("-")}-postgres",
```

Comment:
Hvordan bør vi utlede instanceName. Hvis vi skal ha nummerering som suffix, så blir vi nødt til å kalle herkimer for å se hvilke som allerede er registrert. 
En annen løsning er å bruke deler av hostnavnet. Suffixen i hostnavnet inneholder nummerering, så vi kan kanskje bare bruke dette

Reply:
vi utleder det ved hjelp av hostname for nummerering. Vi bruker også affiliation her